mail_activity_default_assignee/models/mail_activity_type.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+MAGIC_FIELDS = models.MAGIC_COLUMNS
+
+
+class MailActivityType(models.Model):
+    _inherit = "mail.activity.type"
+
+    model_id = fields.Many2one("ir.model", compute="_compute_model_id")
+
+    default_user_field_id = fields.Many2one(
+        "ir.model.fields",
+        domain=f"""
+            [
+                ("name", "not in", {str(MAGIC_FIELDS)}),
+                ("ttype", "in", ["one2many","many2one", "many2many"]),
+                ("model_id", "=", model_id),
```

Comment:
```suggestion
                ("model_id.model", "=", res_model),
```

with this change I think we no longer need the `model_id` field, and so you could drop it to keep it simpler